pkgs/os-specific/linux/apparmor/fix-rc.apparmor.functions.sh

```diff
@@ -1,3 +1,4 @@
+# shellcheck shell=bash
```

Comment:
> alternative suggestion: we enforce a shellcheck check for all changed files and forbid that it is broken.

That would be excellent. What would be an acceptable mechanism for enforcing that? If I understand the documentation, running `pre-commit run --remote-branch ${{ github.event.pull_request.base.ref }}` on a PR should check the shell scripts which have changed in the current PR.

Reply:
Re. your suggested change, why should this script be treated as `sh`?

Reply:
> What would be an acceptable mechanism for enforcing that?

Probably a GitHub Action.

> Re. your suggested change, why should this script be treated as sh?

It is already sh compatible. I am not sure how to handle this properly but not every sh file in the repo is a bash file.

Reply:
> It is already sh compatible.

Doesn't look like it; see in particular the "In POSIX sh, […]" messages:

```
$ shellcheck --shell=sh pkgs/os-specific/linux/apparmor/fix-rc.apparmor.functions.sh

In pkgs/os-specific/linux/apparmor/fix-rc.apparmor.functions.sh line 5:
  $*
  ^-- SC2048: Use "$@" (with quotes) to prevent whitespace problems.


In pkgs/os-specific/linux/apparmor/fix-rc.apparmor.functions.sh line 8:
    aa_log_success_msg $"$STRING "
                       ^---------^ SC3004: In POSIX sh, $".." is undefined.


In pkgs/os-specific/linux/apparmor/fix-rc.apparmor.functions.sh line 10:
    aa_log_failure_msg $"$STRING "
                       ^---------^ SC3004: In POSIX sh, $".." is undefined.


In pkgs/os-specific/linux/apparmor/fix-rc.apparmor.functions.sh line 16:
   [ -n "$1" ] && echo -n $1
                       ^-- SC3037: In POSIX sh, echo flags are undefined.
                          ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
   [ -n "$1" ] && echo -n "$1"


In pkgs/os-specific/linux/apparmor/fix-rc.apparmor.functions.sh line 21:
   [ -n "$1" ] && echo -n $1
                       ^-- SC3037: In POSIX sh, echo flags are undefined.
                          ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
   [ -n "$1" ] && echo -n "$1"


In pkgs/os-specific/linux/apparmor/fix-rc.apparmor.functions.sh line 26:
   [ -n "$1" ] && echo -n $1
                       ^-- SC3037: In POSIX sh, echo flags are undefined.
                          ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
   [ -n "$1" ] && echo -n "$1"


In pkgs/os-specific/linux/apparmor/fix-rc.apparmor.functions.sh line 31:
   [ -n "$1" ] && echo -n $1
                       ^-- SC3037: In POSIX sh, echo flags are undefined.
                          ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
   [ -n "$1" ] && echo -n "$1"

For more information:
  https://www.shellcheck.net/wiki/SC2048 -- Use "$@" (with quotes) to prevent...
  https://www.shellcheck.net/wiki/SC3004 -- In POSIX sh, $".." is undefined.
  https://www.shellcheck.net/wiki/SC3037 -- In POSIX sh, echo flags are undef...
```